File: arrays/find-permutation.cpp
vector<int> Solution::findPerm(const string A, int B) {
    int mini=1, maxi=B;
    vector<int> ans;
    for(int i=0; i<A.size(); i++){
        if(A[i]=='D') ans.push_back(maxi--);
        else ans.push_back(mini++);
    }
    ans.push_back(maxi);
    return ans;
}

File: arrays/find-permutation.py
class Solution:
    def findPerm(self, S, N):
        n = []
        left = 1; right = N

        for x in S:
            if x == 'D':
                n.append(right)
                right -= 1
            else:
                n.append(left)
                left += 1

        n.append(left)
        return n
